### PR TITLE
Fix rewriting of floating point compare statements

### PR DESCRIPTION
**Pull request by ptomin**
Possible fix of floating point compare statements rewriting. I have also added explanation comment.
5 `Cce` tests and 1 `Dfa` test seem to be incorrect. I have fixed it. Hope they are correct now.

### CODE REVIEW

File: src/Arch/X86/X86Rewriter.Fpu.cs [hunk 1]
```diff
@@ -363,30 +363,59 @@ public bool MatchesFstswSequence()
...
+                    jpo    0x05    <
+                    jpo    0x41    <=
+                    jpo    0x44    =
+                    jz     0x01    >=
+                    jz     0x40    !=
+                    jz     0x41    >
+                    jnz    0x01    <
+                    jnz    0x40    =
+                    jnz    0x41    <=
+                */
```
Comment (uxmal) [inline src/Arch/X86/X86Rewriter.Fpu.cs]: Nice documentation! Did you find it online? If so please add a URL.

Comment (ptomin) [inline src/Arch/X86/X86Rewriter.Fpu.cs]: I have combined description of `fcomp`, `fpusw` flags, `jpe/gpo/jz/jz` and created this table

**Comment by uxmal** [comment]
Nice work. Did you fi

**Comment by ptomin** [comment]
This is `intel` documentation https://software.intel.com/sites/default/files/managed/39/c5/325462-sdm-vol-1-2abcd-3abcd.pdf

**Pull request merged**